### PR TITLE
Update to use the new recommended Electron approach

### LINKED ISSUE #36: Unhandled Rejection (TypeError): Cannot read property 'process' of undefined
### Steps to reproduce ###

Everything, like in the [intruction](https://www.imodeljs.org/getting-started/)

1. ```git clone https://github.com/imodeljs/desktop-starter.git```
2. ```npm install```
3. ```npm start```

### Result ###
```
Unhandled Rejection (TypeError): Cannot read property 'process' of undefined
```
Screenshot

### PR DESCRIPTION
**Pull request by calebmshafer**
With the release of iModel.js 2.7.0, the base setup for Electron has changed disallowing `getRemote()` to be used from the frontend.  This is another security measure taken to better secure the client-side code run in the renderer process of Electron.

Instead of using `getRemote` to handle command line args, ipc is directly used to gather the information from the main process, i.e. the backend.

Fixes #36

### CODE REVIEW

**Comment by calebmshafer** [comment]
Waiting on the release of 2.7.4, that will have a fix for running the app directly with electron fixing the following error,

`TypeError: Cannot set property 'allowRendererProcessReuse' of undefined`

**Comment by calebmshafer** [comment]
@williamkbentley this bumps to the latest iModel.js plus fixes some of the issues we have on `master` with the latest changes to the electron config.

@jffmarker I'd like to create a new release based on this once it merges.  What testing should I do to make sure it will work with the iTwin Snapshot app? Assuming that it will auto-update.

**Comment by jffmarker** [comment]
> @jffmarker I'd like to create a new release based on this once it merges. What testing should I do to make sure it will work with the iTwin Snapshot app? Assuming that it will auto-update.

I'm not sure how to e2e test it... but start by building the package locally (`npx run-s lint build:prod build:dist`), which should result in a ZIP file. Extract to a new directory (ZIP itself has files in the root) and try to (a) double-click 'Desktop Starter.exe' to see it open to a blank view, and (b) from a shell pass it a single arg, the full path to a snapshot BIM file, and see that it opens a view to that file.

**Comment by calebmshafer** [comment]
> > @jffmarker I'd like to create a new release based on this once it merges. What testing should I do to make sure it will work with the iTwin Snapshot app? Assuming that it will auto-update.
> 
> I'm not sure how to e2e test it... but start by building the package locally (`npx run-s lint build:prod build:dist`), which should result in a ZIP file. Extract to a new directory (ZIP itself has files in the root) and try to (a) double-click 'Desktop Starter.exe' to see it open to a blank view, and (b) from a shell pass it a single arg, the full path to a snapshot BIM file, and see that it opens a view to that file.

@jffmarker ah okay, I did `a` before pushing but haven't tried out `b`. I'll test that out now.

**Review by Josh-Schifter — APPROVED** [review]

**Comment by calebmshafer** [comment]
Ran into issue with parsing the command line, almost wrapped up the fix. Doing some additional testing and then I'll push up the fix.

**Comment by calebmshafer** [comment]
@jffmarker I added the path handling parts you wrote to the backend and it's now passed to the frontend via the `getConfig` ipc.  Can you take a quick look before we merge and release this?

File: src/backend/main.ts
Comment (jffmarker) [inline src/backend/main.ts:86]: - why is this async?
- why can it only be called once?

Comment (calebmshafer) [reply src/backend/main.ts:86]: > * why is this async?

Doesn't technically need to be but shouldn't hurt anything either.  The more we allow the event loop to free up the more responsive the frontend will be.

> * why can it only be called once?

We only call it once during initialization but I can switch it since you may want to look at the environment variables set in the backend again.

Comment (jffmarker) [reply src/backend/main.ts:86]: But without any awaits, I don't think you're actually freeing up the loop. But okay.

My concern was more about the comment... this sounded like you were artificially limiting this. If that's not the case, then I'm not sure you need a comment at all?

Comment (calebmshafer) [reply src/backend/main.ts:86]: Good call on the comment.  I'll remove it.

**Pull request merged**
